crates/python/src/qpu/result_data.rs

```diff
@@ -43,4 +43,40 @@ impl PyQpuResultData {
     fn readout_values(&self, py: Python<'_>) -> PyResult<HashMap<String, PyReadoutValues>> {
         self.as_inner().readout_values().to_python(py)
     }
+
+    fn to_raw_readout_data(&self, py: Python<'_>) -> RawQpuReadoutData {
+        RawQpuReadoutData {
+            mappings: self.as_inner().mappings().clone(),
+            readout_values: self
+                .as_inner()
+                .readout_values()
+                .iter()
+                .map(|(register, values)| {
+                    (
+                        register.to_string(),
+                        match values {
+                            ReadoutValues::Integer(values) => PyList::new(py, values).into_py(py),
+                            ReadoutValues::Real(values) => PyList::new(py, values).into_py(py),
+                            ReadoutValues::Complex(values) => PyList::new(py, values).into_py(py),
+                        },
+                    )
+                })
+                .collect::<HashMap<String, Py<PyList>>>(),
+        }
+    }
+}
+
+#[derive(Debug)]
+#[pyclass(name = "RawQPUReadoutData")]
+pub struct RawQpuReadoutData {
+    #[pyo3(get)]
+    mappings: HashMap<String, String>,
+    #[pyo3(get)]
+    readout_values: HashMap<String, Py<PyList>>,
+}
```

Comment:
The idea here is to make it easier to get at what Python users likely want from `QPUResultData`. The type safety provided by `QpuResultData` and `ReadoutValues` is nice, but only if you care about it. If you don't, it's cumbersome to have to call `inner()` on every instance of `ReadoutValues`.

Reply:
Maybe put this in a doc comment?